sdk/src/main/java/com/mapbox/maps/renderer/FpsManager.kt

```diff
@@ -0,0 +1,213 @@
+package com.mapbox.maps.renderer
+
+import android.os.Handler
+import androidx.annotation.WorkerThread
+import androidx.core.os.postDelayed
+import com.mapbox.maps.logI
+import com.mapbox.maps.logW
+import kotlin.math.pow
+
+@WorkerThread
+internal class FpsManager(
+  private val handler: Handler
+) {
+  private var userRefreshRate = USER_DEFINED_REFRESH_RATE_NOT_SET
+  private var userToScreenRefreshRateRatio: Double? = null
+
+  private var screenRefreshRate = SCREEN_METRICS_NOT_DEFINED
+  private var screenRefreshPeriodNs = -1L
+  private var previousFrameTimeNs = -1L
+  private var previousDrawnFrameIndex = 0
+  private var frameRenderTimeAccumulatedNs = 0L
+
+  private var preRenderTimeNs = -1L
+  private var choreographerTicks = 0
+  private var choreographerSkips = 0
+
+  internal var fpsChangedListener: OnFpsChangedListener? = null
+
+  fun setScreenRefreshRate(screenRefreshRate: Int) {
+    if (this.screenRefreshRate == screenRefreshRate) {
+      return
+    }
+    this.screenRefreshRate = screenRefreshRate
+    screenRefreshPeriodNs = ONE_SECOND_NS / screenRefreshRate
+    if (userRefreshRate != USER_DEFINED_REFRESH_RATE_NOT_SET) {
+      userToScreenRefreshRateRatio = userRefreshRate.toDouble() / screenRefreshRate
```

Comment:
In the doc, we mentioned 
> When setting this value higher than screen physically supports - max possible screen FPS rate will be used.
So in theory the `userToScreenRefreshRateRatio` shouldn't exceed 1.0, maybe we can cap the `userToScreenRefreshRateRatio` below 1.0?

```suggestion
      userToScreenRefreshRateRatio = (userRefreshRate.toDouble() / screenRefreshRate).coerceIn(0.0, 1.0)
```